lib/index.js

```diff
@@ -0,0 +1,55 @@
+import createLogger from './logs';
+import createKey from './key';
+import pandaValidate from 'pan-domain-validate-user';
+
+const AUTH_FAILED_ERROR_MESSAGE = 'panda-auth-failed';
+const COOKIE_NAME = 'gutoolsAuth-assym';
+
+export default function(config, overrideHttps, validate = pandaValidate) {
+    // For backward compatibility, config can be the domain string or an object
+    if (typeof config === 'string') {
+        config = { domain: config };
+    }
+    const {
+        domain,
+        refresh = 10 * 60 * 1000, // 10 minutes
```

Comment:
Although we update the key in the background - from what I can see here the key update is triggered when it has expired when being accessed? Should be fine either way in reality.